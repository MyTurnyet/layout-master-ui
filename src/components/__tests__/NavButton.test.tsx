import React from 'react';
import {render, screen} from '@testing-library/react';
import StyledNavButton, {NavButtonProp} from "components/NavButton";
import {MemoryRouter} from "react-router-dom";
import {library} from "@fortawesome/fontawesome-svg-core";
import {fas} from "@fortawesome/free-solid-svg-icons";
import {ThemeProvider} from "styled-components";
import defaultTheme from "../../theme/theme";

describe(' NavButton ', () => {
    beforeAll(() => {
        library.add(fas);
    })
    it('should render not active label, icon and point to url ', () => {
        const buttonProps: NavButtonProp = {
            path: './foo',
            label: 'Testing',
            icon: 'home'
        }
        render(
            <MemoryRouter>
                <ThemeProvider theme={defaultTheme}>
                    <StyledNavButton {...buttonProps} isActive={false}/>
                </ThemeProvider>
            </MemoryRouter>);
        const link = screen.getByRole('link', {name: "Testing"});
        expect(link).not.toHaveClass('active')
        expect(link).toBeInTheDocument()
    });
    it('should render active label, icon and point to url ', () => {
        const buttonProps: NavButtonProp = {
            path: './foo',
            label: 'Testing',
            icon: 'home'
        }
        render(
            <MemoryRouter>
                <ThemeProvider theme={defaultTheme}>
                    <StyledNavButton {...buttonProps} isActive={true}/>
                </ThemeProvider>
            </MemoryRouter>);
        screen.debug()
        const link = screen.getByRole('link', {name: "Testing"});
        // expect(link.parentElement).toHaveClass('active')
        expect(link).toBeInTheDocument()
    });
});
