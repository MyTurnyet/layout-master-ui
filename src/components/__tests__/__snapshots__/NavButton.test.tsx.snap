// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` NavButton  should render active label, icon and point to url  1`] = `
.c2 {
  font-size: 12px;
  text-transform: capitalize;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 20px;
  background: #FFFFFF;
}

.c1 svg {
  -webkit-filter: drop-shadow(3px 3px 3px rgba(0,0,0,.4));
  filter: drop-shadow(3px 3px 3px rgba(0,0,0,.4));
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  cursor: pointer;
}

.c0 a {
  color: #FA4141;
}

<div
  class="c0"
  label="Testing"
>
  <a
    href="/foo"
  >
    <div
      class="c1"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-home fa-w-18 "
        data-icon="home"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
          fill="currentColor"
        />
      </svg>
    </div>
    <span
      class="c2"
    >
      Testing
    </span>
  </a>
</div>
`;

exports[` NavButton  should render not active label, icon and point to url  1`] = `
.c2 {
  font-size: 12px;
  text-transform: capitalize;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 20px;
  background: #FFFFFF;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  cursor: pointer;
}

.c0 a {
  color: #A1A1A1;
}

<div
  class="c0"
  label="Testing"
>
  <a
    href="/foo"
  >
    <div
      class="c1"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-home fa-w-18 "
        data-icon="home"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
          fill="currentColor"
        />
      </svg>
    </div>
    <span
      class="c2"
    >
      Testing
    </span>
  </a>
</div>
`;
